src/startup/IndexModeCLIParser.cpp

```diff
@@ -0,0 +1,180 @@
+                                    forbidIndexWriteoutSwitch->getValue(),
+                                    dictCompressionSwitch->getValue());
+
+    if (storeForDecompressedBlocksArg->isSet()) {
+        runner->enableWriteOutOfDecompressedBlocksAndStatistics(storeForDecompressedBlocksArg->getValue());
+        ErrorAccumulator::always("Storage location for decompressed blocks: '",
+                                 storeForDecompressedBlocksArg->getValue(), "'");
+    }
+
+    if (storeForPartialDecompressedBlocksArg->isSet()) {
+        ErrorAccumulator::always("Storage location for partial decompressed block data file to: '",
+                                 storeForPartialDecompressedBlocksArg->getValue(), "'");
+        runner->enableWriteOutOfPartialDecompressedBlocks(storeForPartialDecompressedBlocksArg->getValue());
+    }
+    return runner;
+}
+
+tuple<_StringValueArg, shared_ptr<ValuesConstraint<string>>>
+IndexModeCLIParser::createSelectIndexEntryStorageStrategyArg(CmdLine *cmdLineParser) const {
+    vector<string> allowedMetrics{"ByteDistance", "BlockDistance"};
+    auto allowedMetricsConstraint = make_shared<ValuesConstraint<string>>(allowedMetrics);
+
+    auto arg = make_shared<ValueArg<string>>(
+            "m", "selectIndexEntryMetric",
+            string("Selects the metric which is used to calculate, which compressed block will be referenced with ") +
+            "index entry in the resulting FQI file. The default value is \"ByteDistance\".",
+            false,
+            "ByteDistance", allowedMetricsConstraint.get(), *cmdLineParser);
+    return {arg, allowedMetricsConstraint};
+}
+
+_IntValueArg IndexModeCLIParser::createBlockIntervalArg(CmdLine *cmdLineParser) const {
+    return _makeIntValueArg(
+            "b",
+            "blockInterval",
+            string("Defines, that an index entry will be written to the index file for every nth compressed ") +
+            "block. Value needs to be in range from [1 .. n].",
+            false,
+            -1, cmdLineParser);
+}
+
+_SwitchArg IndexModeCLIParser::createDictCompressionSwitchArg(CmdLine *cmdLineParser) const {
+    return _makeSwitchArg(
+            "c", "disableDictionaryCompression",
+            string("Disable compression of the stored dictionaries. By default, the stored dictionaries will") +
+            " be compressed using zlib and this will usually decrease the index size to 1/3rd to 1/4th "
+            "compared to an uncompressed index.",
+            cmdLineParser, true);
+}
+
+_SwitchArg IndexModeCLIParser::createForbidIndexWriteoutSwitchArg(CmdLine *cmdLineParser) const {
+    return _makeSwitchArg(
+            "F", "forbidIndexWrite",
+            string("Forbids writing the index file. More interesting for debugging."),
+            cmdLineParser);
+}
+
+_SwitchArg IndexModeCLIParser::createDisableFailsafeDistanceSwitchArg(CmdLine *cmdLineParser) const {
+    return _makeSwitchArg(
+            "S", "disableFailsafeDistance",
```

Comment:
Consider renaming, dependent on the outcome of our discussion on the name of this concept.